lcls-twincat-common-components/Library/Devices/PPM/FB_PPM_PowerMeter.TcPOU

```diff
@@ -28,61 +41,145 @@ VAR
     fVoltageBuffer: ARRAY[1..1000] OF LREAL;
 
     {attribute 'pytmc' := '
-        pv: CALIB
+        pv: STC
         io: input
     '}
-    fCalibBase: LREAL;
+    fbTempSensor: FB_CC_TempSensor;
 
+    // Used to get beam mode and rate
     {attribute 'pytmc' := '
-        pv: CALIB_BUFFER
-        io: input
+        pv: AccTypeFromEpics
+        link: TPR:FEES:GD02:0:MODE.RVAL
+    '}
+    uAccelerator: UINT;
+    {attribute 'pytmc' := '
+        pv: NCBeamRateFromEpics
+        link: TPG:SYS0:1:DST03:RATE
+    '}
+    uSC_HXR_Frequency: UINT;
+    {attribute 'pytmc' := '
+        pv: SCBeamRateFromEpics
+        link: TPG:SYS0:1:DST04:RATE
     '}
-    fCalibBaseBuffer: ARRAY[1..1000] OF LREAL;
+    uSC_SXR_Frequency: UINT;
+    // Placeholders until I find the rights pvs for normal conducting
+        {attribute 'pytmc' := '
+        pv: NCBeamRateFromEpics
+        link: TPG:SYS0:1:DST03:RATE
+    '}
+    uNC_HXR_Frequency: UINT;
+    {attribute 'pytmc' := '
+        pv: SCBeamRateFromEpics
+        link: TPG:SYS0:1:DST04:RATE
+    '}
+    uNC_SXR_Frequency: UINT;
```

Comment:
I think the `pv: ` lines in this block might have some typos/copy-paste errors since some of them are the same